src/Agent.Worker/AgentPluginManager.cs

```diff
@@ -160,18 +160,36 @@ public async Task RunPluginTaskAsync(IExecutionContext context, string plugin, D
 
             // construct plugin context
             var target = context.StepTarget();
+            Variables.TranslationMethod translateToHostPath = Variables.DefaultStringTranslator;
+
+            ContainerInfo containerInfo = target as ContainerInfo;
+            // Since plugins run on the host, but the inputs and variables have already been translated
+            // to the container path, we need to convert them back to the host path
+            // TODO: look to see if there is a better way to not have translate these back
+            if (containerInfo != null)
+            {
+                var newInputs = new Dictionary<string,string>();
+                foreach (var var in inputs)
+                {
+                    newInputs[var.Key] = containerInfo.TranslateToHostPath(var.Value);
+                }
+                inputs = newInputs;
```

Comment:
Any reason we're creating a new dictionary here? E.g. couldn't we just do:
```
foreach (var var in inputs)
{
    inputs[var.Key] = containerInfo.TranslateToHostPath(var.Value);
}
```

Reply:
I thought I tried it, but in C# got an error an about modifying a list while iterating over it.

Reply:
Hm, that makes sense. I think you might be able to do:
```
foreach (var key in inputs.Keys)
{
    inputs[key] = containerInfo.TranslateToHostPath(inputs[key]);
}
```

Reply:
that worked!